rule/var_naming.go

```diff
@@ -176,6 +135,51 @@ func (*VarNamingRule) Name() string {
 	return "var-naming"
 }
 
+func (r *VarNamingRule) applyPackageCheckRules(file *lint.File, onFailure func(failure lint.Failure)) {
+	fileDir := path.Dir(file.Name)
+
+	// Protect pkgsWithNameFailure from concurrent modifications
+	r.pkgNameAlreadyChecked.Lock()
+	defer r.pkgNameAlreadyChecked.Unlock()
+	defer r.pkgNameAlreadyChecked.add(fileDir) // mark this package as already checked
+
+	// Do we need to proceed with the checks on package name?
+	mustSkipCheck := r.pkgNameAlreadyChecked.has(fileDir) || r.skipPackageNameChecks
+	if mustSkipCheck {
+		return
+	}
+
+	pkgNameNode := file.AST.Name
+	pkgName := pkgNameNode.Name
+	pkgNameLower := strings.ToLower(pkgName)
+	if _, ok := r.extraBadPackageNames[pkgNameLower]; ok {
+		onFailure(r.pkgNameFailure(pkgNameNode, "avoid bad package names"))
+		return
+	}
+
+	if _, ok := defaultBadPackageNames[pkgNameLower]; ok {
+		onFailure(r.pkgNameFailure(pkgNameNode, "avoid meaningless package names"))
+		return
+	}
+
+	// Package names need slightly different handling than other names.
+	if strings.Contains(pkgName, "_") && !strings.HasSuffix(pkgName, "_test") {
+		onFailure(r.pkgNameFailure(pkgNameNode, "don't use underscores in package names"))
```

Comment:
I propose reverting changes on failure messages (because it may break someone who relies on it):
```suggestion
		onFailure(r.pkgNameFailure(pkgNameNode, "don't use an underscore in package name"))
```

Better is to create a separate PR for this.

Reply:
Relying on text messages is known to be fragile. Someone doing it should expect breaks. Anyway, I think we need to gain freedom to modify/fix failure messages. If we stick to "don't modify msg because someone might rely on it" we will not have that freedom. IMO the way to fix this is to evolve failures to have a machine-friendly _failure code_. Like that, messages (for human consumption) can be freely modified.

Reply:
I’m not totally against modifying the messages - I agree that relying on text is inherently fragile.

My only concern is about traceability: it might be better to handle message changes in a separate PR. That way, anyone looking through the history can clearly see when and why a message was changed. It just makes things a bit easier to follow down the line.